addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/FetchMetadataRequestHeadersScanRule.java

```diff
@@ -0,0 +1,303 @@
+                        "embed",
+                        "empty",
+                        "font",
+                        "frame",
+                        "iframe",
+                        "image",
+                        "manifest",
+                        "object",
+                        "paintworklet",
+                        "report",
+                        "script",
+                        "serviceworker",
+                        "sharedworker",
+                        "style",
+                        "track",
+                        "video",
+                        "worker",
+                        "xslt");
+
+        SecFetchDest(Supplier<AlertBuilder> newAlert) {
+            super(newAlert, "-3");
+        }
+
+        @Override
+        AlertBuilder build(HttpRequestHeader httpRequestHeaders) {
+            AlertBuilder alert = null;
+            List<HttpHeaderField> httpHeaderFields = httpRequestHeaders.getHeaders();
+
+            boolean hasSecFetchDest =
+                    httpHeaderFields.stream()
+                            .anyMatch(httpHeaderField -> httpHeaderField.getName().equals(HEADER));
+
+            if (!hasSecFetchDest) {
+                alert = this.alert("missing.", "");
+            } else {
+                List<String> value = httpRequestHeaders.getHeaderValues(HEADER);
+                if (!VALID_VALUES.contains(value.get(0))) {
+                    alert = this.alert("invalid-values.", value.get(0));
+                }
+            }
+            return alert;
+        }
+
+        @Override
+        protected String getHeader() {
+            return HEADER;
+        }
+
+        @Override
+        protected String getString(String param) {
+            return Constant.messages.getString(SFD_PREFIX_MESSAGE + param);
+        }
+    }
+
+    static class SecFetchUser extends FetchMetaDataRequestHeaders {
+        private static final String HEADER = "Sec-Fetch-User";
+        private static final String SFU_PREFIX_MESSAGE =
+                FETCH_METADATA_REQUEST_MESSAGE_PREFIX + "sfu.";
+
+        SecFetchUser(Supplier<AlertBuilder> newAlert) {
+            super(newAlert, "-4");
+        }
+
+        @Override
+        AlertBuilder build(HttpRequestHeader httpRequestHeaders) {
+            AlertBuilder alert = null;
+            List<HttpHeaderField> httpHeaderFields = httpRequestHeaders.getHeaders();
+
+            boolean hasSecFetchUser =
+                    httpHeaderFields.stream()
+                            .anyMatch(httpHeaderField -> httpHeaderField.getName().equals(HEADER));
+
+            if (!hasSecFetchUser) {
+                alert = this.alert("missing.", "");
+            } else {
+                List<String> value = httpRequestHeaders.getHeaderValues(HEADER);
+                if (!(value.get(0)).equals("?1")) {
+                    alert = this.alert("invalid-values.", value.get(0));
+                }
+            }
+            return alert;
+        }
```

Comment:
The first three of these seem the same so may as well be the default implementation and thus deduplicated, this one only differs in a single line, could it be passed somehow?

Reply:
Yeah sure if that works, I didn't notice, but seems reasonable to me.